staging/src/k8s.io/component-base/metrics/registry.go

```diff
@@ -17,12 +17,36 @@ limitations under the License.
 package metrics
 
 import (
+	"sync"
+	"sync/atomic"
+
 	"github.com/blang/semver"
 	"github.com/prometheus/client_golang/prometheus"
 	dto "github.com/prometheus/client_model/go"
+
 	apimachineryversion "k8s.io/apimachinery/pkg/version"
 )
 
+var (
+	showHiddenOnce sync.Once
+	showHidden     atomic.Value
```

Comment:
why is this not a property on the registry?

Reply:
My initial iteration actually did that but I ended up changing my mind.

My reasoning was this: we are not guaranteed to have a single registry per component (kubelet already doesn't), if we were to have this as a property on the registry, we would either need to have an individual flag added per registry for each component, (i.e. `--show-deprecated-hidden-metrics-for-endpoint-1` and `--show-deprecated-hidden-metrics-for-endpoint-2`), rather a single flag for that component (i.e. `--show-deprecated-hidden-metrics`). It felt reasonable to trade the ability to configure individual registries with keeping the API surface area a bit simpler but provide a single option to configure all registries for that component, but that is definitely a subjective call. What do you think?

Reply:
I’d prefer not to introduce more global state and instead pass it into each registry as a parameter, which is not mutually exclusive with having a single flag to do so.